src/dynamics/ads_player/ads_controlbar/ads_controlbar.js

```diff
@@ -32,6 +32,8 @@ Scoped.define("module:Ads.Dynamics.Controlbar", [
 
                 channels: {
                     "ads:adProgress": function(event) {
+                        if (!this.get("duration") || this.get("duration") == 0)
+                            this.set("duration", event.getAdData().duration);
```

Comment:
Hi @ibnutri, the duration comes from the AdsPlayer dynamic, I don't think we should be setting it here:
https://github.com/betajs/betajs-media-components/blob/master/src/dynamics/ads_player/ads_player.html#L64

Reply:
I just tested your other PR for the missing ad icon and it works perfect even without this change, so I think we can close this one. 🙂
You did a great job on this one, it was a good idea to create a separate template for showing just the items we want to show. 🙂